### DISCUSSION TITLE
Combining Branch + Command

### DISCUSSION THREAD

**Discussion by phillip-haydon** [spectreconsole/spectre.console#1305] [General]
**Describe the bug**
I was playing around trying to see if I could combine a command and a branch, so I could do a 'default' command on the branch...

i.e I want to be able to have

app.exe status
app.exe status service
app.exe status apps

Where status is a branch but if `service` or `apps` isnt passed in, it would default to `service`.

What actually happens is you end up with 2 commands listed for `status` the command varient and the branch varient. But only the command varient can be executed. 

So I'm not sure if you would call it a bug, or something that spectre should throw up on and tell me I'm dumb when I try to run it. 

**To Reproduce**

```
  config.AddCommand<TestCommand>("status");

  config.AddBranch("status", b =>
  {
    b.AddCommand<TestCommand>("service");
    b.AddCommand<TestCommand>("apps");
  });
```

**Expected behavior**
I'm not sure what to expect.

Either:

1) It allows a command and branch with the same name, however this would cause downstream issues if the command has positional arguments. 
2) Throw up and tell me I'm a twat and not to stupid things when I try to run it.

**Comment by nils-a** [comment]
@phillip-haydon Branches _may_ have a default, or they _may not_. 
I feel, what you are missing is `SetDefaultCommand`.

What you describe can be achieved by

```csharp
config.AddBranch("status", b =>
{
  b.SetDefaultCommand<ServiceCommand>();
  b.AddCommand<ServiceCommand>("service");
  b.AddCommand<AppsCommand>("apps");
});
```
